plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts

```diff
@@ -69,39 +76,50 @@ export class FactRetrieverEngine {
       factRetrieverRegistry,
       factRetrieverContext,
       factRetrieverContext.logger,
+      scheduler,
       defaultCadence,
+      defaultTimeout,
     );
   }
 
   schedule() {
     const registrations = this.factRetrieverRegistry.listRegistrations();
     const newRegs: string[] = [];
-    registrations.forEach(registration => {
-      const { factRetriever, cadence, lifecycle } = registration;
-      if (!this.scheduledJobs.has(factRetriever.id)) {
-        const cronExpression =
-          cadence || this.defaultCadence || randomDailyCron();
-        if (!validate(cronExpression)) {
-          this.logger.warn(
-            `Validation failed for cron expression ${cronExpression} when trying to schedule fact retriever ${factRetriever.id}`,
-          );
-          return;
-        }
-        const job = schedule(
-          cronExpression,
-          this.createFactRetrieverHandler(factRetriever, lifecycle),
+    registrations.forEach(async registration => {
+      const { factRetriever, cadence, lifecycle, timeout } = registration;
+      const cronExpression =
+        cadence || this.defaultCadence || randomDailyCron();
+      try {
+        // disabling eslint here so we can parse the CronTime and verify it's properly formatted
+        // eslint-disable-next-line no-new
+        new CronTime(cronExpression);
+      } catch {
+        this.logger.warn(
```

Comment:
Hm the scheduler already checks this. Maybe wrap the `this.scheduler.scheduleTask` in a try/catch, and in the catch rethrow it like

```ts
catch(e) {
  throw new Error(`Failed to schedule fact retriever ${factRetriever.id}, ${e}`);
```

or something like that? Would that even mean that this package can stop depending on `cron` entirely?

Reply:
Aw sweet, good move